src/daemon/mod.rs

```diff
@@ -286,159 +412,164 @@ pub(super) async fn start(
-
-    // Initialize ChainMuxer
+        services,
+    )?);
     let chain_muxer = ChainMuxer::new(
-        Arc::clone(&state_manager),
-        peer_manager.clone(),
+        state_manager.clone(),
+        p2p_service.peer_manager().clone(),
         mpool.clone(),
-        network_send.clone(),
-        network_rx,
-        Arc::new(Tipset::from(&genesis_header)),
-        tipset_sender.clone(),
-        tipset_receiver,
+        p2p_service.network_sender().clone(),
+        p2p_service.network_receiver(),
+        Arc::new(Tipset::from(
+            state_manager.chain_store().genesis_block_header(),
+        )),
         opts.stateless,
     )?;
-    let bad_blocks = chain_muxer.bad_blocks_cloned();
-    let sync_state = chain_muxer.sync_state_cloned();
-    let sync_network_context = chain_muxer.sync_network_context();
+    Ok(chain_muxer)
+}
+
+fn start_chain_muxer_service(
+    services: &mut JoinSet<anyhow::Result<()>>,
+    chain_muxer: ChainMuxer<DbType, MpoolRpcProvider<DbType>>,
+) {
     services.spawn(async { Err(anyhow::anyhow!("{}", chain_muxer.await)) });
+}
 
+async fn start_health_check_service_if_needed(
+    services: &mut JoinSet<anyhow::Result<()>>,
+    config: &Config,
+    state_manager: &StateManager<DbType>,
+    p2p_service: &Libp2pService<DbType>,
+    chain_muxer: &ChainMuxer<DbType, MpoolRpcProvider<DbType>>,
+) -> anyhow::Result<()> {
     if config.client.enable_health_check {
         let forest_state = crate::health::ForestState {
             config: config.clone(),
-            chain_config: chain_config.clone(),
-            genesis_timestamp: genesis_header.timestamp,
-            sync_state: sync_state.clone(),
-            peer_manager,
-            settings_store: chain_store.settings(),
+            chain_config: state_manager.chain_config().clone(),
+            genesis_timestamp: state_manager.chain_store().genesis_block_header().timestamp,
+            sync_state: chain_muxer.sync_state().clone(),
+            peer_manager: p2p_service.peer_manager().clone(),
+            settings_store: state_manager.chain_store().settings(),
         };
-
         let listener =
             tokio::net::TcpListener::bind(forest_state.config.client.healthcheck_address).await?;
-
         services.spawn(async move {
             crate::health::init_healthcheck_server(forest_state, listener)
                 .await
                 .context("Failed to initiate healthcheck server")
         });
     }
+    Ok(())
+}
 
-    // Start services
+#[allow(clippy::too_many_arguments)]
+fn start_rpc_and_f3_services_if_needed(
+    services: &mut JoinSet<anyhow::Result<()>>,
+    opts: &CliOpts,
+    config: &Config,
+    keystore: Arc<RwLock<KeyStore>>,
+    state_manager: &Arc<StateManager<DbType>>,
+    chain_muxer: &ChainMuxer<DbType, MpoolRpcProvider<DbType>>,
+    start_time: chrono::DateTime<chrono::Utc>,
+    network_name: String,
+    p2p_peer_id: PeerId,
+    admin_jwt: String,
+    shutdown: mpsc::Sender<()>,
```

Comment:
It might be worth zipping it into a single context structure to simplify the method signature (and future additions).

Reply:
Splitted into `maybe_start_rpc_service` and `maybe_start_f3_service` with less parameters for each.